docs/contrib/workflow.md

```diff
@@ -0,0 +1,232 @@
+# Contribution to OpenKilda project
+
+Anyone can contribute in OpenKilda project. This instruction describes how
+OpenKilda contribution looks like: from an idea to OpenKilda release.
+
+# Workflow
+
+## Idea
+
+If you are willing to add a new feature or make an improvement it makes sense
+to discuss it with other contributors first. To find current active contributors
+you can open a list of [open pull requests](https://github.com/telstra/open-kilda/pulls)
+and pick an author of recent pull request to write him/her a message about your
+idea. He or she will invite you to a place for idea discussions.
+
+If you don't have an idea, but you still want to contribute - please take a look on a
+list of [open issues](https://github.com/telstra/open-kilda/issues). If you find
+an open and unassigned issue (of course you will) you can assign it to yourself and
+start to work on it.
+
+### Design
+
+If you are working on a small feature, fix or improvement you can skip the design step.
+But if you are working on a big feature, especially which changes API, you must
+create a design document first. Examples of design documents can be found [here](../design).
+
+## Development
+
+### Git branches
+
+To make any changes in OpenKilda repository you need to create a git branch.
+There are 2 main branches in the repository:
+
+* `develop` - contains changes which were not yet released
+* `master` - contains last [release](https://github.com/telstra/open-kilda/releases)
+
+Please checkout your branch from `develop` branch.
+
+### Branch name
+
+There are no strict branch name restrictions. The only rule is to make branch name meaningful.
+In practice, we are using the following prefixes for branches:
+
+* `doc/` - for adding/updating the documentation
+* `feature/` - for adding new features
+* `fix/` - for fixing current codebase
+* `test/` - for tests
+* `release` - for releases (only a release manager can create such branches)
+* `hotfix` - for hotfixes which must be released ASAP (only a release manager can create such branches)
+
+After the prefix there must be a very short description of a changes.
```

Comment:
I'd prefer that branch name start from issue ID and to be as close to issue description as possible (of course, trying to make it short). E.g. `feature/1535-add-wednesday-reminder`